### PR TITLE
ownCloud 9.0 rework/optional improvments.

### PR DESCRIPTION
**Pull request by ypid**
- [x] Update test-suite. For now check: https://travis-ci.org/ypid/ansible-owncloud

Related to: #28

### CODE REVIEW

**Comment by ypid** [comment]
Tests are not passing because https://github.com/debops/ansible-owncloud/commit/3c977a6d50fa03b7b4fc4e169718cb26d65a7e0b has been merged into master and the auto merge which travis is really testing includes this commit. Tests should otherwise be passing and I will ensure this before the merge by updating the test-suite.

Current tests can be found here: https://travis-ci.org/ypid/ansible-owncloud

File: tasks/system_package_management.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+# .. vim: foldmarker=[[[,]]]:foldmethod=marker
+
+- name: Get ownCloud APT OpenPGP key
+  apt_key:
+    url: '{{ owncloud__apt_repo_key_url }}'
```
Comment (drybjed) [inline tasks/system_package_management.yml]: The GPG key should be installed from an OpenPGP key server, not from the same source as the APT repository.

Comment (ypid) [inline tasks/system_package_management.yml]: Thanks for the review! I think I already tried that a few months ago and just retired. The public key is not on keyservers it seems: gpg --recv-keys BCECA90325B072AB1245F739AB7C32C35180350A

I don’t see a direct problem there. Would you prefer to include the public key with the role?

Comment (drybjed) [inline tasks/system_package_management.yml]: I think that GPG keys should be pulled from a different source than the APT repository information, that way the trust can be estabilished independently from the source.

I would try and convince the ownCloud authors to publish their GPG keys on the keyservers. They are already in circulation, publication in the OpenPGP keyserver network will make access to them easier not only for DebOps, but for other projects that wish to authenticate the packages as well.

If they won't do it, then yes, GPG keys should be included in the role to provide independent verifable source of the key material. At least this sounds better than downloading the GPG key from a random server on the Internet which can vanish at any moment.

The key should be copied from the role's `files/` directory to the remote host to say, `/usr/local/src/owncloud/` and installed via `apt_key` module. This should ensure idempotent operation.

Comment (ypid) [inline tasks/system_package_management.yml]: I included the pubkey for now until this is fixed upstream. Thanks.

Tracked upstream: [ownCloud](https://github.com/owncloud/core/issues/25710), [NextCloud](https://github.com/nextcloud/server/issues/755)

**Pull request merged**